pippy/PipelineSchedule.py

```diff
@@ -187,8 +187,12 @@ def step(self, microbatches):
                     dist.batch_isend_irecv(ops)
 
```

Comment:
adding torch.cuda.synchronize() after dist.batch_isend_irecv() here would stop the forward for ranks 1,2,3 to be in sync with rank 0